File: SafeVote/check_db_status.js
const mongoose = require('mongoose');

const MONGO_URI = "mongodb+srv://stharunkumar069_db_user:siet%40123@cluster0.frcnaxx.mongodb.net/safevote?retryWrites=true&w=majority&appName=Cluster0";

async function check() {
    try {
        await mongoose.connect(MONGO_URI);
        console.log("Connected to MongoDB.");

        const db = mongoose.connection.db;
        const collections = await db.listCollections().toArray();

        console.log("\nCollections in 'safevote' database:");
        for (let col of collections) {
            const count = await db.collection(col.name).countDocuments();
            console.log(`- ${col.name}: ${count} documents`);
        }

        process.exit(0);
    } catch (err) {
        console.error(err);
        process.exit(1);
    }
}

check();

File: SafeVote/backend/elections/management/commands/check_db_status.py
from django.core.management.base import BaseCommand
from utils.database import get_db, check_db_connection

class Command(BaseCommand):
    help = "Inspect MongoDB connection and collection document counts matching check_db_status.js"

    def handle(self, *args, **options):
        self.stdout.write("Connecting to MongoDB...")
        if not check_db_connection():
            self.stderr.write(self.style.ERROR("[FAIL] Failed to connect to MongoDB!"))
            return

        self.stdout.write(self.style.SUCCESS("[OK] Connected to MongoDB."))
        db = get_db()
        collections = db.list_collection_names()

        self.stdout.write(f"\nCollections in '{db.name}' database:")
        for col_name in collections:
            count = db[col_name].count_documents({})
            self.stdout.write(f"- {col_name}: {count} documents")
